pkg/validate/security_context_linux.go

```diff
@@ -317,6 +318,53 @@ var _ = framework.KubeDescribe("Security Context", func() {
 			Expect(groups).To(ContainElement("5678"))
 		})
 
+		It("runtime should add SupplementalGroups to the groups which container's primary UID belongs to in the container image when defined", func() {
+			By("create pod")
+			podID, podConfig = framework.CreatePodSandboxForContainer(rc)
+
+			supplementalGroups := []int64{1234, 5678}
+			By("create container for security context SupplementalGroups")
+			containerName := "container-with-SupplementalGroups-and-predefined-group-image-test-" + framework.NewUUID()
+			logPath := fmt.Sprintf("%s.log", containerName)
+			containerConfig := &runtimeapi.ContainerConfig{
+				Metadata: framework.BuildContainerMetadata(containerName, framework.DefaultAttempt),
+				Image:    &runtimeapi.ImageSpec{Image: testImagePreDefinedGroup},
+				Command:  []string{"sh", "-c", "id -G"},
+				Linux: &runtimeapi.LinuxContainerConfig{
+					SecurityContext: &runtimeapi.LinuxContainerSecurityContext{
+						SupplementalGroups: supplementalGroups,
+					},
+				},
+				LogPath: logPath,
+			}
+			containerID := framework.CreateContainer(rc, ic, containerConfig, podID, podConfig)
```

Comment:
[the image (gcr.io/k8s-staging-cri-tools/test-image-predefined-group)](https://github.com/kubernetes-sigs/cri-tools/blob/master/images/image-predefined-group/Dockerfile) is
- default user is `default-user(uid=1000)`
- the user belongs to `group-defined-in-image(gid=50000)` group in the image

Thus, the supplementary group of the container processes(`id -G`) should be:

```console
$ id -G
1000 1234 5678 50000

# 1000: self
# 1234 5678: SupplementalGroups
# 50000: defined in the container for uid=1000
```